fix cast str to byte tensor for pytorch

hoshimura opened the pull request:
## 🚀 🚀 Pull Request

### Checklist:

- [x ]  [My code follows the style guidelines of this project](https://www.notion.so/activeloop/Engineering-Guidelines-d6e502306d0e4133a8ca507516d1baab) and the [Contributing document](https://github.com/activeloopai/Hub/blob/release/2.0/CONTRIBUTING.md)
- [ ]  I have commented my code, particularly in hard-to-understand areas
- [x]  I have kept the `coverage-rate` up
- [x]  I have performed a self-review of my own code and resolved any problems
- [x]  I have checked to ensure there aren't any other open [Pull Requests](https://github.com/activeloopai/Hub/pulls) for the same change
- [ ]  I have described and made corresponding changes to the relevant documentation
- [x]  New and existing unit tests pass locally with my changes


### Changes

- fixes #1312

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/activeloopai/Hub?pullRequest=1313) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/activeloopai/Hub?pullRequest=1313) before we can accept your contribution.<br/><hr/>**Johan Sternby** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/activeloopai/Hub?pullRequest=1313) it.</sub>

hoshimura commented:
I don't understand why the complex tests fail. The transform function should to my knowledge receive a tensor, not a numpy.ndarray

hoshimura commented:
@farizrahman4u I guess you did not like my attempt to support strings in pytorch by casting them to a byte tensor? In case it was just a merge error, I readded them in the 7th commit. Please elaborate on what can be done if not splicing the string to bytes is an ok workaround!

hoshimura commented:
the windows failing test is not related to my changes from what I can see

aliubimov commented:
We refactored major part of pytorch implementation. Prefetch LRU was replaced with new `Dataset` dedicated for PyTorch integration. Cast of numpy to a torch tensor is now moved [to this function](https://github.com/activeloopai/Hub/blob/main/hub/integrations/pytorch/dataset.py#L42)

hoshimura commented:
> We refactored major part of pytorch implementation. Prefetch LRU was replaced with new `Dataset` dedicated for PyTorch integration. Cast of numpy to a torch tensor is now moved [to this function](https://github.com/activeloopai/Hub/blob/main/hub/integrations/pytorch/dataset.py#L42)

Excellent!, Now my changes seem superfluous. Keeping the tests just in case

AbhinavTuli commented:
Hey @hoshimura! I'm closing this PR as we've implemented another fix for this (that doesn't involve conversion to and from bytes and works without a transform), see #1342 
I've included your test in the PR. Thanks for bringing attention to this bug!
